### PR TITLE
[NAT-517] Change TaggingService for FHIR version support

### PR DESCRIPTION
**Pull request by bitPogo**
## Description
This adds the FHIR version support for `fetchRecords` and `countRecords`.

## Motivation and Context
Currently `fetchRecords` and `countRecords` of the RecordService do not support versions of FHIR, which they need to.

## How is it being implemented?
`getTagFromType` of the TaggingService now adds, if it encounters a FhirResource, the version of the resource. Therefore the underlying FhirElementHelper got a new function - `resolveFhirVersion`.

## How Has This Been Tested?
The tests of the TaggingService and functional tests (aka Integration) had been adjusted. `resolveFhirVersion` got its own tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -23,6 +23,7 @@ toc::[]
 * `EncryptedRecord`, `EncryptedKey`, `EncryptedKeyTypeAdapter` to Kotlin
 * CryptoService#encryptString -> CryptoService#encryptAndEncodeString
 * CryptoService#decryptString -> CryptoService#decodeAndDecryptString
+* TaggingService#getTagFromType now tags now FhirResourceType and FhirVersion
```
Comment (wmontwe) [inline CHANGELOG.adoc:26]: one now too much

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -44,6 +45,7 @@ toc::[]
 * RecordService#checkDataRestrictions for Resources with unextractable Attachments
 * missing Task in SDKContract for some of the operations
 * all Errors to be reported to our Logger
+* RecordService#fetchRecords had been mixed up parameter.
```
Comment (wmontwe) [inline CHANGELOG.adoc:48]: does this belong here?

File: sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt [hunk 3]
```diff
@@ -19,7 +19,7 @@ package care.data4life.sdk.fhir
 import care.data4life.crypto.GCKey
 
 //TODO: make internal
-interface FhirContract {
+class FhirContract {
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt:22]: why is this changed to a class?

File: sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt [hunk 4]
```diff
@@ -30,4 +30,10 @@ interface FhirContract {
                 encryptedResource: String
         ): T
     }
+
+    enum class FhirVersion(val version: String) {
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt:34]: 👍

File: sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt [hunk 5]
```diff
@@ -30,4 +30,10 @@ interface FhirContract {
...
+        FHIR_3("3.0.1"),
+        FHIR_4("4.0.1"),
+        NO_FHIR("invalid")
```
Comment (wmontwe) [inline sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt:37]: UNKNOWN("unknown") is more commonly used

**Review by wmontwe — CHANGES_REQUESTED** [review]
👍 

just some small things

File: sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt (hunk 3 above)
Comment (bitPogo) [reply sdk-core/src/main/java/care/data4life/sdk/fhir/FhirContract.kt]: You right, forgot to change it back - originally I wanted the enum to `internal`, which currently does not work. However as we have it in another place - interfaces do not support those kind of visibility modifiers.

**Review by wmontwe — APPROVED** [review]
👍
